server/src/api/customers.ts

```diff
@@ -39,4 +40,22 @@ customerRouter.get(
   }
 );
 
+/**
+ * Create a customer if they do not already exist.
+ */
+customerRouter.post(
+  '/',
+  async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
Another additional thing is that if you are already getting customer in the previous execution (customerRouter.get...), are you able to make judgement on whether the customer exists or not based on previous call? If so, you may avoid the addMerchant call if the customer is already known to exist.

Reply:
@xlai20 As mentioned in the comment above, these are just specifying how the router will respond to these requests, and not executed. So this code block in here will execute independently of the get request.